feg/gateway/fabfile.py

```diff
@@ -11,125 +11,320 @@
 limitations under the License.
 """
 import sys
+from typing import Dict, List
 
-from fabric.api import cd, env, get, lcd, local, puts, run
-from fabric.utils import abort
+import urllib3
+from fabric.api import hide, lcd, local
 
 sys.path.append('../../orc8r')
 import tools.fab.dev_utils as dev_utils  # NOQA
-import tools.fab.pkg as pkg  # NOQA
-from tools.fab.hosts import split_hoststring # NOQA
-from tools.fab.vagrant import setup_env_vagrant  # NOQA
+import tools.fab.types as types
 
-AWS = 'aws --region eu-west-1'
-DEFAULT_CERT = "$MAGMA_ROOT/.cache/test_certs/rootCA.pem"
+NETWORK_ID = 'feg_test'
 
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def register_feg():
+    _register_federation_network()
+    _register_feg()
+
+
+class RadiusConfig:
+    def __init__(self,
+                 DAE_addr: str = '127.0.0.1:3799',
+                 acct_addr: str = '127.0.0.1:1813',
+                 auth_addr: str = '127.0.0.1:1812',
+                 network: str = 'udp',
+                 secret: str = 'MTIzNDU2'):
+        self.DAE_addr = DAE_addr
+        self.acct_addr = acct_addr
+        self.auth_addr = auth_addr
+        self.network = network
+        self.secret = secret
+
+
+class AAAServerConfig:
+    def __init__(self,
+                 accounting_enabled: bool = True,
+                 create_session_on_auth: bool = True,
+                 event_logging_enabled: bool = False,
+                 idle_session_timeout_ms: int = 21600000,
+                 radius_config: RadiusConfig = RadiusConfig()):
+        self.accounting_enabled = accounting_enabled
+        self.create_session_on_auth = create_session_on_auth
+        self.event_logging_enabled = event_logging_enabled
+        self.idle_session_timeout_ms = idle_session_timeout_ms
+        self.radius_config = radius_config
+
+
+class EapAkaTimeout:
+    def __init__(self,
+                 challenge_ms: int = 20000, error_notification_ms: int = 10000,
+                 session_authenticated_ms: int = 5000,
+                 session_ms: int = 43200000):
+        self.challenge_ms = challenge_ms
+        self.error_notification_ms = error_notification_ms
+        self.session_authenticated_ms = session_authenticated_ms
+        self.session_ms = session_ms
+
+
+class EapAkaConfig:
+    def __init__(self, plmn_ids: List[str] = None,
+                 timeout: EapAkaTimeout = EapAkaTimeout()):
+        if plmn_ids is None:
+            plmn_ids = ['123456']
+        self.plmn_ids = plmn_ids
+        self.timeout = timeout
+
```

Comment:
do we need to add the definition for the 'new' `eap-sim`?

```
  eap_sim:
    type: object
    description: eap_sim configuration
    properties:
      timeout:
        $ref: '#/definitions/eap_sim_timeouts'
      plmn_ids:
        type: array
        items:
          type: string
          minLength: 5
          maxLength: 6
          pattern: '^(\d{5,6})$'
          example: '1234
```

Reply:
It looks like it's not a required member of the struct so I left it out